wasi/wasi_test.go

```diff
@@ -254,14 +263,30 @@ func TestArgsGetAPIReturnError(t *testing.T) {
 		tc := tt
 
 		t.Run(tc.name, func(t *testing.T) {
-			errno := wasiEnv.args_get(hostFunctionCallContext, tc.argsPtr, tc.argsBufPtr)
-			require.Equal(t, EINVAL, errno)
+			ret, _, err := store.CallFunction("test", "args_get", uint64(tc.argsPtr), uint64(tc.argsBufPtr))
+			require.NoError(t, err)
+			require.Equal(t, uint64(EINVAL), ret[0]) // ret[0] is returned errno
 		})
 	}
 }
 
-func buildHostFunctionCallContext() *wasm.HostFunctionCallContext {
-	return &wasm.HostFunctionCallContext{
-		Memory: &wasm.MemoryInstance{Buffer: make([]byte, wasm.MemoryPageSize), Min: 1},
+func instantiateWasmStore(wasmFile string, wasiEnv *WASIEnvironment) (*wasm.Store, error) {
+	buf, err := os.ReadFile(wasmFile)
+	if err != nil {
+		return nil, err
+	}
+	mod, err := wbinary.DecodeModule(buf)
+	if err != nil {
+		return nil, err
+	}
+	store := wasm.NewStore(interpreter.NewEngine())
+	err = wasiEnv.Register(store)
+	if err != nil {
+		return nil, err
+	}
+	err = store.Instantiate(mod, "test")
```

Comment:
pass the name of the module as a var then in your tests, you can use `moduleName` which will be obviously same as here

Reply:
ah nice catch! thanks, I've updated it.